apps/src/code-studio/assets/DefaultSpritesEditor.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import {
+  getDefaultList,
+  updateDefaultList
+} from '@cdo/apps/assetManagement/animationLibraryApi';
+import DefaultSpriteRow from '@cdo/apps/code-studio/assets/DefaultSpriteRow';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
+
+export default class DefaultSpritesEditor extends React.Component {
+  state = {
+    isLoading: true,
+    defaultList: {}, // Dictionary with name as key and sprite object as value
+    pendingChanges: 0,
+    isUpdating: false
+  };
+
+  componentDidMount() {
+    getDefaultList()
+      .then(spriteDefault => {
+        var spriteList = {};
+        spriteDefault['default_sprites'].map(
+          sprite => (spriteList[sprite.name] = sprite)
+        );
+        this.setState({defaultList: spriteList, isLoading: false});
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  deleteSpriteFromDefaults = spriteName => {
+    delete this.state.defaultList[spriteName];
```

Comment:
this is directly changing state, which will have unintended side-effects. all state should be updated through `this.setState` (there are some useful guidelines [here](https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly)). the way i usually see this done in an immutable way is something like:

```javascript
let defaultList = {...this.state.defaultList}; // copy the original state
delete defaultList[spriteName]; // mutate the copy

this.setState(state => {
  defaultList,
  pendingChanges: state.pendingChanges + 1
});
```

Reply:
Ha dang, I fixed this forward, but never ported it back. Thanks for catching!